File: lib_common/src/main/java/com/ebook/common/mvvm/model/BaseModel.java
package com.ebook.common.mvvm.model;

import android.app.Application;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;


public abstract class BaseModel implements IBaseModel {
    protected Application mApplication;
    private CompositeDisposable mCompositeDisposable;
    public BaseModel(Application application) {
        mApplication = application;
        mCompositeDisposable = new CompositeDisposable();
    }
    public void addSubscribe(Disposable disposable) {
        if (mCompositeDisposable == null) {
            mCompositeDisposable = new CompositeDisposable();
        }
        mCompositeDisposable.add(disposable);
    }

    @Override
    public void onCleared() {
        if (mCompositeDisposable != null) {
            mCompositeDisposable.clear();
        }
    }

}

File: lib_common/src/main/java/com/xrn1997/common/mvvm/model/BaseModel.kt
package com.xrn1997.common.mvvm.model

import android.app.Application
import com.xrn1997.common.mvvm.IBaseModel
import io.reactivex.rxjava3.disposables.CompositeDisposable
import io.reactivex.rxjava3.disposables.Disposable

/**
 *BaseModel 用于实现IBaseModel接口
 * @author xrn1997
 */
@Suppress("unused")
abstract class BaseModel(protected var mApplication: Application) : IBaseModel {
    private val _compositeDisposable: CompositeDisposable = CompositeDisposable()
    val mCompositeDisposable get() = _compositeDisposable

    override fun addSubscribe(disposable: Disposable) {
        _compositeDisposable.add(disposable)
    }

    override fun onCleared() {
        _compositeDisposable.clear()
    }

}